GeneralsMD/Code/GameEngine/Include/GameNetwork/NetPacketStructs.h

```diff
@@ -0,0 +1,415 @@
+////////////////////////////////////////////////////////////////////////////////
+
+struct NetPacketRunAheadCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketFrameField frame;
+	NetPacketPlayerIdField playerId;
+	NetPacketCommandIdField commandId;
+	NetPacketDataFieldHeader dataHeader;
+	UnsignedShort runAhead;
+	UnsignedByte frameRate;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Destroy Player Command Packet
+////////////////////////////////////////////////////////////////////////////////
+
+struct NetPacketDestroyPlayerCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketFrameField frame;
+	NetPacketPlayerIdField playerId;
+	NetPacketCommandIdField commandId;
+	NetPacketDataFieldHeader dataHeader;
+	UnsignedInt playerIndex;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Keep Alive Command Packet
+////////////////////////////////////////////////////////////////////////////////
+
+struct NetPacketKeepAliveCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Disconnect Keep Alive Command Packet
+////////////////////////////////////////////////////////////////////////////////
+
+struct NetPacketDisconnectKeepAliveCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Disconnect Player Command Packet
+////////////////////////////////////////////////////////////////////////////////
+
+struct NetPacketDisconnectPlayerCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketCommandIdField commandId;
+	NetPacketDataFieldHeader dataHeader;
+	UnsignedByte slot;
+	UnsignedInt disconnectFrame;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Packet Router Command Packets
+////////////////////////////////////////////////////////////////////////////////
+
+// Packet router query command packet
+struct NetPacketRouterQueryCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+// Packet router ack command packet
+struct NetPacketRouterAckCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Base Packed Struct for All Command Messages
+// Contains the minimum common fields that all command messages share
```

Comment:
This comment is not true. Considering the messages do not share common packed structures atop, I think none of the packed structs should use inheritance. If they all had consistent headers, then inheritance would be good.